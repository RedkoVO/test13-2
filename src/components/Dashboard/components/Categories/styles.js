export default theme => ({
  root: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    flexWrap: 'wrap',
    width: '100%',

    [theme.breakpoints.down('md')]: {
      justifyContent: 'center'
    }
  },
  categories: {
    display: 'flex',
    flexDrection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    flexWrap: 'wrap',
    width: '405px',
    marginTop: '60px',

    [theme.breakpoints.down('lg')]: {
      width: '390px'
    },
    [theme.breakpoints.down('md')]: {
      width: '620px'
    },
    [theme.breakpoints.down('sm')]: {
      maxWidth: '630px',
      width: '390px'
    }
  },
  categTitleWrap: {
    width: '100%',
    paddingLeft: '50%',
    boxSizing: 'border-box'
  },
  categoriesTitle: {
    display: 'block',
    fontSize: '26px',
    fontWeight: 700,
    color: '#575757',
    textAlign: 'center',
    marginBottom: '0px',
    textDecoration: 'none',
    margin: '0 auto',
    transform: 'translateX(-50%)'
  }
})
