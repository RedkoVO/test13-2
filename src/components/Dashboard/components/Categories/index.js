import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import withStyles from '@material-ui/core/styles/withStyles'

import CategorieItem from './components/CategorieItem'

import styles from './styles'

const Categories = ({ classes, categories }) => (
  <div className={classes.root}>
    {categories.map(item => (
      <div className={classes.categories} key={item.id}>
        <div className={classes.categTitleWrap}>
          <Link to="/" className={classes.categoriesTitle}>Categories:</Link>
        </div>

        {item.game.map(itemGames => (
          <CategorieItem game={itemGames} key={itemGames.id} />
        ))}
      </div>
    ))}
  </div>
)

Categories.propTypes = {
  classNamees: PropTypes.object,
  categories: PropTypes.array
}

export default withStyles(styles)(Categories)