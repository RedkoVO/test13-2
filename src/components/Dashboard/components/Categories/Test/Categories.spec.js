import * as React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'

import Categories from '../'

const data = [
  {
    id: 1,
    game: [
      {
        id: 1,
        title: 'Most popular',
        name: '4Clover',
        img: 'CloverLogo1',
        bundle: '4clover'
      },
      {
        id: 2,
        title: 'Most popular',
        name: 'Dice games',
        img: 'CloverLogo1',
        bundle: '4clover'
      },
      {
        id: 3,
        title: 'For children',
        name: 'Word games',
        img: 'CloverLogo1',
        bundle: '4clover'
      },
      {
        id: 4,
        title: 'For children',
        name: 'Funny games',
        img: 'CloverLogo1',
        bundle: '4clover'
      },
      {
        id: 5,
        title: 'For children',
        name: 'Word games',
        img: 'CloverLogo1',
        bundle: '4clover'
      },
      {
        id: 6,
        title: 'For children',
        name: 'Word games',
        img: 'CloverLogo1',
        bundle: '4clover'
      }
    ]
  },
  {
    id: 2,
    game: [
      {
        id: 1,
        title: 'Most popular',
        name: 'Loto games',
        img: 'CloverLogo1',
        bundle: '4clover'
      },
      {
        id: 2,
        title: 'Most popular',
        name: 'Dice games',
        img: 'CloverLogo1',
        bundle: '4clover'
      },
      {
        id: 3,
        title: 'For children',
        name: 'Word games',
        img: 'CloverLogo1',
        bundle: '4clover'
      },
      {
        id: 4,
        title: 'For children',
        name: 'Funny games',
        img: 'CloverLogo1',
        bundle: '4clover'
      },
      {
        id: 5,
        title: 'For children',
        name: 'Word games',
        img: 'CloverLogo1',
        bundle: '4clover'
      },
      {
        id: 6,
        title: 'For children',
        name: 'Word games',
        img: 'CloverLogo1',
        bundle: '4clover'
      }
    ]
  }
]

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter keyLength={0}>
        <Categories categories={data} />
      </MemoryRouter>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
