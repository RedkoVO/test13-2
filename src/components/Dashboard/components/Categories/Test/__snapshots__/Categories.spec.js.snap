// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Categories-root-1"
>
  <div
    className="Categories-categories-2"
  >
    <div
      className="Categories-categTitleWrap-3"
    >
      <a
        className="Categories-categoriesTitle-4"
        href="/"
        onClick={[Function]}
      >
        Categories:
      </a>
    </div>
    <div
      className="CategorieItem-root-5"
    >
      <a
        className="CategorieItem-gameImg-6"
        href="/game/4clover"
        onClick={[Function]}
      >
        <img
          alt=""
          className="CategorieItem-gameImgLink-7"
          src="CloverLogo1"
        />
      </a>
      <a
        className="CategorieItem-gameLink-8"
        href="/game/4clover"
        onClick={[Function]}
      >
        Most popular
        : 
        <br />
         
        <span>
          4Clover
        </span>
      </a>
    </div>
    <div
      className="CategorieItem-root-5"
    >
      <a
        className="CategorieItem-gameImg-6"
        href="/game/4clover"
        onClick={[Function]}
      >
        <img
          alt=""
          className="CategorieItem-gameImgLink-7"
          src="CloverLogo1"
        />
      </a>
      <a
        className="CategorieItem-gameLink-8"
        href="/game/4clover"
        onClick={[Function]}
      >
        Most popular
        : 
        <br />
         
        <span>
          Dice games
        </span>
      </a>
    </div>
    <div
      className="CategorieItem-root-5"
    >
      <a
        className="CategorieItem-gameImg-6"
        href="/game/4clover"
        onClick={[Function]}
      >
        <img
          alt=""
          className="CategorieItem-gameImgLink-7"
          src="CloverLogo1"
        />
      </a>
      <a
        className="CategorieItem-gameLink-8"
        href="/game/4clover"
        onClick={[Function]}
      >
        For children
        : 
        <br />
         
        <span>
          Word games
        </span>
      </a>
    </div>
    <div
      className="CategorieItem-root-5"
    >
      <a
        className="CategorieItem-gameImg-6"
        href="/game/4clover"
        onClick={[Function]}
      >
        <img
          alt=""
          className="CategorieItem-gameImgLink-7"
          src="CloverLogo1"
        />
      </a>
      <a
        className="CategorieItem-gameLink-8"
        href="/game/4clover"
        onClick={[Function]}
      >
        For children
        : 
        <br />
         
        <span>
          Funny games
        </span>
      </a>
    </div>
    <div
      className="CategorieItem-root-5"
    >
      <a
        className="CategorieItem-gameImg-6"
        href="/game/4clover"
        onClick={[Function]}
      >
        <img
          alt=""
          className="CategorieItem-gameImgLink-7"
          src="CloverLogo1"
        />
      </a>
      <a
        className="CategorieItem-gameLink-8"
        href="/game/4clover"
        onClick={[Function]}
      >
        For children
        : 
        <br />
         
        <span>
          Word games
        </span>
      </a>
    </div>
    <div
      className="CategorieItem-root-5"
    >
      <a
        className="CategorieItem-gameImg-6"
        href="/game/4clover"
        onClick={[Function]}
      >
        <img
          alt=""
          className="CategorieItem-gameImgLink-7"
          src="CloverLogo1"
        />
      </a>
      <a
        className="CategorieItem-gameLink-8"
        href="/game/4clover"
        onClick={[Function]}
      >
        For children
        : 
        <br />
         
        <span>
          Word games
        </span>
      </a>
    </div>
  </div>
  <div
    className="Categories-categories-2"
  >
    <div
      className="Categories-categTitleWrap-3"
    >
      <a
        className="Categories-categoriesTitle-4"
        href="/"
        onClick={[Function]}
      >
        Categories:
      </a>
    </div>
    <div
      className="CategorieItem-root-5"
    >
      <a
        className="CategorieItem-gameImg-6"
        href="/game/4clover"
        onClick={[Function]}
      >
        <img
          alt=""
          className="CategorieItem-gameImgLink-7"
          src="CloverLogo1"
        />
      </a>
      <a
        className="CategorieItem-gameLink-8"
        href="/game/4clover"
        onClick={[Function]}
      >
        Most popular
        : 
        <br />
         
        <span>
          Loto games
        </span>
      </a>
    </div>
    <div
      className="CategorieItem-root-5"
    >
      <a
        className="CategorieItem-gameImg-6"
        href="/game/4clover"
        onClick={[Function]}
      >
        <img
          alt=""
          className="CategorieItem-gameImgLink-7"
          src="CloverLogo1"
        />
      </a>
      <a
        className="CategorieItem-gameLink-8"
        href="/game/4clover"
        onClick={[Function]}
      >
        Most popular
        : 
        <br />
         
        <span>
          Dice games
        </span>
      </a>
    </div>
    <div
      className="CategorieItem-root-5"
    >
      <a
        className="CategorieItem-gameImg-6"
        href="/game/4clover"
        onClick={[Function]}
      >
        <img
          alt=""
          className="CategorieItem-gameImgLink-7"
          src="CloverLogo1"
        />
      </a>
      <a
        className="CategorieItem-gameLink-8"
        href="/game/4clover"
        onClick={[Function]}
      >
        For children
        : 
        <br />
         
        <span>
          Word games
        </span>
      </a>
    </div>
    <div
      className="CategorieItem-root-5"
    >
      <a
        className="CategorieItem-gameImg-6"
        href="/game/4clover"
        onClick={[Function]}
      >
        <img
          alt=""
          className="CategorieItem-gameImgLink-7"
          src="CloverLogo1"
        />
      </a>
      <a
        className="CategorieItem-gameLink-8"
        href="/game/4clover"
        onClick={[Function]}
      >
        For children
        : 
        <br />
         
        <span>
          Funny games
        </span>
      </a>
    </div>
    <div
      className="CategorieItem-root-5"
    >
      <a
        className="CategorieItem-gameImg-6"
        href="/game/4clover"
        onClick={[Function]}
      >
        <img
          alt=""
          className="CategorieItem-gameImgLink-7"
          src="CloverLogo1"
        />
      </a>
      <a
        className="CategorieItem-gameLink-8"
        href="/game/4clover"
        onClick={[Function]}
      >
        For children
        : 
        <br />
         
        <span>
          Word games
        </span>
      </a>
    </div>
    <div
      className="CategorieItem-root-5"
    >
      <a
        className="CategorieItem-gameImg-6"
        href="/game/4clover"
        onClick={[Function]}
      >
        <img
          alt=""
          className="CategorieItem-gameImgLink-7"
          src="CloverLogo1"
        />
      </a>
      <a
        className="CategorieItem-gameLink-8"
        href="/game/4clover"
        onClick={[Function]}
      >
        For children
        : 
        <br />
         
        <span>
          Word games
        </span>
      </a>
    </div>
  </div>
</div>
`;
